mypy_django_plugin/transformers/fields.py

```diff
@@ -150,6 +152,22 @@ def set_descriptor_types_for_field(
         is_set_nullable=is_set_nullable or is_nullable,
         is_get_nullable=is_get_nullable or is_nullable,
     )
+
+    # reconcile set and get types with the base field class
+    base_field_type = next(base for base in default_return_type.type.mro if base.fullname == FIELD_FULLNAME)
+    mapped_instance = map_instance_to_supertype(default_return_type, base_field_type)
+    mapped_set_type, mapped_get_type = mapped_instance.args
+
+    # bail if either mapped_set_type or mapped_get_type have type Never
+    if not (isinstance(mapped_set_type, UninhabitedType) or isinstance(mapped_get_type, UninhabitedType)):
+        # only replace set_type and get_type with mapped types if their original value is Any
```

Comment:
Perhaps we just want to always replace if the mapped types are not Any.

This is actually a use-case I need where I use a `SmallInteger` field to represent a custom enum type so I'll have to set `_pyi_private_set_type` and `_pyi_private_get_type` to `Any` in my code.

Reply:
I don't see any harm in supporting what you mention, don't think it'll affect anything else

Reply:
Okay, I have changed this so explicit type params will override the `_pyi_private_set_type` and `_pyi_private_get_type` variables